demos/eigenvalues_QG_basinmodes/qgbasinmodes.rst

```diff
@@ -0,0 +1,209 @@
+derivations in Pedlosky:1992 and Vallis:2006.
+This model consists of an evolution equation
+for the Potential Vorticity, :math:`q`, and an elliptic problem through
+which we can determine the streamfunction,
+
+.. math::
+
+   \partial_{t}q + \vec{\nabla}\cdot (\vec{u}q) + \beta v = 0 \\
+   q = \nabla^{2} \psi - F\psi
+
+Where :math:`\psi` is the stream-function, :math:`\vec{u}=(u, v)` is the
+velocity field, q is the Potential Vorticity (PV), :math:`\beta` is the
+Coriolis parameter and F is the rotational Froude number. The velocity
+field is easily obtained using
+
+.. math::
+
+   \vec{u} = \vec{\nabla}^{\bot}\psi,
+   \quad \mbox{ with } \quad
+   \vec{\nabla}^{\bot} =  \hat{e_{z}} \times \vec{\nabla}.
+
+We assume that the amplitude of the wave motion is very small, which
+allows us to linearize the equations of motion and therefore neglect the
+nonlinear advection,
+
+.. math:: \frac{\partial}{\partial t} (\nabla^{2} \psi - F\psi) = - \beta \frac{\partial \psi}{\partial x}
+
+We look for wave-like solutions that are periodic in time, with a
+frequency of :math:`\omega`
+
+.. math:: \psi = \hat{\psi}(x, y)e^{-i\omega t}.
+
+This has the advantage of removing the time derivative from the equation
+and replaceing it with an eigenvalue, :math:`i \omega`, By substituting
+the above solution into the QG equation, we can find a complex
+eigenvalue problem of the form
+
+.. math:: i\omega (\nabla^{2} \hat{\psi} - F\hat{\psi}) = \hat{\beta} \frac{\partial \hat{\psi}}{\partial x}
+
+Weak Formulation
+----------------
+
+To use a finite element method it is necessary to formulate the weak
+form and then we can use SLEPc in Firedrake to compute eigenvalue
+problems easily.
+To begin, we multiply this equation by a Test Function :math:`\phi`
+and integrate over the domain :math:`A`. 
+
+.. math::
+
+   i\omega \iint_{A} \Big(\phi\cdot\nabla^{2} \hat{\psi}\,dA - F\phi\hat{\psi}\,dA\Big) = \hat{\beta}\iint_{A} \phi \cdot \frac{\partial \hat{\psi}}{\partial x}\,dA
+
+To remove the Laplacian operator we use integration by parts and the Divergence theorem to obtain
+
+  .. math::
+
+     \iint_{A} \phi \cdot \nabla^{2}\hat{\psi} \,dA = - \iint_{A} \nabla\phi \cdot \nabla\hat{\psi}\,dA + \oint_{\partial A} \phi \cdot \frac{\partial \hat{\psi}}{\partial n} \,dS
+
+No-normal flow boundary conditions are required and mathematically this
+means that the streamfunction must be a constant on the boundary. Since
+the test functions inheret these boundary condition,
+:math:`\hat \phi = 0` on the boundary, the boundary integral
+vanishes and the weak form becomes,
+
+.. math::
+
+   i\omega \iint_{A} \Big( \nabla\phi\cdot\nabla \hat{\psi}\,dA + F\phi\hat{\psi}\Big)\,dA = \hat{\beta}\iint_{A} \phi \cdot \frac{\partial \hat{\psi}}{\partial x}\,dA
+
+Firedrake code
+--------------
+
+Using this form, we can now implement this eigenvalue problem in
+Firedrake. We import the Firedrake, PETSc, and SLEPc. ::
```

Comment:
If using "the", then we should not treat Firedrake, PETSc, and SLEPc as proper nouns, so better to write either:

```
We import the Firedrake, PETSc, and SLEPc libraries.
```

Or

```
We import Firedrake, PETSc, and SLEPc.
```